src/main.ts

```diff
@@ -922,21 +936,25 @@ function generateEncode(ctx: Context, fullName: string, messageDesc: DescriptorP
         const tag = ((field.number << 3) | 2) >>> 0;
         const toNumber = getEnumMethod(typeMap, field.typeName, 'ToNumber');
         chunks.push(code`
-          writer.uint32(${tag}).fork();
-          for (const v of message.${fieldName}) {
-            writer.${toReaderCall(field)}(${toNumber}(v));
+          if (message.${fieldName} !== undefined && message.${fieldName}.length !== 0) {
```

Comment:
Unlike the other spot where I suggested `|| []`, it does seem nice here to keep the tag completely off the wire if there are no elements in the collection.